### ISSUE TITLE
panic: Cannot seek beyond end of blob

### ISSUE THREAD

**Issue by ahmetb** [paulmey/inspect-azure-vhd#1]
Tried this with a VHD of a running VM and got this error:

```
./inspect-azure-vhd [MY_VHD_SAS_URI]
Reading partition table...
Inspecting ext4 filesystem on first partition...
Inode count:     1925120
Block count:     32984249329331968
Block size:      4096
Cluster size:    4
Blocks/group:    32768
Clusters/group:  32768
Inode/group:     8192
Magic:           61267
State:           1
FeatureCompat:   HasJournal|ExtAttr|ResizeInodes|DirIndex(0x0000003c)
FeatureIncompat: Filetype|Recoder|Extents|FlexBG(0x00000246)
FeatureROCompat: SparseSuper|LargeFile|HugeFile|GDTCsum|DirNlink|ExtraIsize(0x0000007b)
Iterating /var/log, downloading interesting files...
panic: Cannot seek beyond end of blob

goroutine 1 [running]:
main.main()
    /Users/alp/workspace/inspect-azure-vhd/src/github.com/paulmey/inspect-azure-vhd/main.go:60 +0x53e

goroutine 25 [IO wait]:
net.(*pollDesc).Wait(0xc208010bc0, 0x72, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010bc0, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208010b60, 0xc20828e000, 0x800, 0x800, 0x0, 0x512b48, 0xc20828a678)
    /usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20803e0c8, 0xc20828e000, 0x800, 0x800, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/net/net.go:121 +0xdc
crypto/tls.(*block).readFromUntil(0xc208271ad0, 0x513f80, 0xc20803e0c8, 0x5, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/crypto/tls/conn.go:454 +0xe6
crypto/tls.(*Conn).readRecord(0xc2080b5600, 0x17, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/crypto/tls/conn.go:539 +0x2da
crypto/tls.(*Conn).Read(0xc2080b5600, 0xc2082e2000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/crypto/tls/conn.go:904 +0x166
net/http.noteEOFReader.Read(0x51d450, 0xc2080b5600, 0xc2082c5e98, 0xc2082e2000, 0x1000, 0x1000, 0xc208034200, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20812d860, 0xc2082e2000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
    <autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208310840)
    /usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc208310840, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc2082c5e40)
    /usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:660 +0xc9f

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
    /usr/local/Cellar/go/1.4.2/libexec/src/runtime/asm_amd64.s:2232 +0x1

goroutine 26 [select]:
net/http.(*persistConn).writeLoop(0xc2082c5e40)
    /usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:661 +0xcbc
```

**Comment by paulmey** [comment]
Cool, I bet it's not my test vhd... :-) Let me get an uri for your vhd and fix this...

**Comment by paulmey** [comment]
Fixed in d027175a75ef2991b6bdbfa5df737cd575ad337f

**Comment by ahmetb** [comment]
confirming the fix.
